src/Extension/FluentSearchFormExtension.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * SilverStripe core search work around when using Fluent
+ * @link https://gist.github.com/baukezwaan/266c469758319daea5460f2a5647c525
+ */
+
+namespace TractorCow\Fluent\Extension;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Search\SearchForm;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\SS_List;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Simple extension to enable SS Core search to work with SS Fluent
+ */
+class FluentSearchFormExtension extends SearchForm
+{
+    /**
+     * Return dataObjectSet of the results using current request to get info from form.
+     * Wraps around {@link searchEngine()}.
+     *
+     * @return SS_List
+     */
+    public function getResults()
+    {
+        // Get request data from request handler
+        $request = $this->getRequestHandler()->getRequest();
+
+        $keywords = $request->requestVar('Search');
+
+        $andProcessor = function ($matches) {
+            return ' +' . $matches[2] . ' +' . $matches[4] . ' ';
+        };
+
+        $notProcessor = function ($matches) {
+            return ' -' . $matches[3];
+        };
+
+        $keyword_patterns = [
+            'andProcessor' => '/()("[^()"]+")( and )("[^"()]+")()/i',
+            'andProcessor' => '/(^| )([^() ]+)( and )([^ ()]+)( |$)/i',
+            'notProcessor' => '/(^| )(not )("[^"()]+")/i',
```

Comment:
You have duplicate array keys here; You'll end up with a 2-length array not a 4-length one.

Instead, try this

```php
$patterns = [
    [
        'pattern' => '/()("[^()"]+")( and )("[^"()]+")()/i',
        'callback' => $andProcessor,
    ],
    // etc...
];
```